[24.0.0] ndmp part fails build on Fedora-40

### Bareos component version

24.0.0

### Steps to reproduce

Try to build it.

### Expected results

working build

### Actual results

GCC error:
```
[ 46%] Building C object core/src/ndmp/CMakeFiles/ndmjob.dir/ndmjob_job.c.o
cd /builddir/build/BUILD/bareos-Release-24.0.0/redhat-linux-build/core/src/ndmp && /usr/bin/ccache base_dir=/builddir/build/BUILD/bareos-Release-24.
0.0 hash_dir=true namespace=bareos /usr/lib64/ccache/gcc -DHAVE_GCC=1 -D_FILE_OFFSET_BITS=64 -I/usr/include/tirpc -I/builddir/build/BUILD/bareos-Rel
ease-24.0.0/core/src -I/builddir/build/BUILD/bareos-Release-24.0.0/redhat-linux-build/core/src -I/builddir/build/BUILD/bareos-Release-24.0.0/redhat-
linux-build/core/src/ndmp -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_
SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/red
hat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-poin
ter -mno-omit-leaf-frame-pointer -fdebug-prefix-map=/builddir/build/BUILD/bareos-Release-24.0.0=/usr/src/bareos -fmacro-prefix-map=/builddir/build/B
UILD/bareos-Release-24.0.0=/usr/src/bareos -Werror -Wall -Wextra -Wshadow -DNDMOS_OPTION_NO_TEST_AGENTS -DNDMOS_CONST_VENDOR_NAME=\"Bareos\ GmbH\ \&
\ Co.KG\" -DNDMOS_CONST_PRODUCT_NAME=\"Bareos\" -DHAVE_LINUX_OS -fPIE -MD -MT core/src/ndmp/CMakeFiles/ndmjob.dir/ndmjob_job.c.o -MF CMakeFiles/ndmj
ob.dir/ndmjob_job.c.o.d -o CMakeFiles/ndmjob.dir/ndmjob_job.c.o -c /builddir/build/BUILD/bareos-Release-24.0.0/core/src/ndmp/ndmjob_job.c
../../../../core/src/ndmp/ndmjob_args.c: In function ‘process_args’:
../../../../core/src/ndmp/ndmjob_args.c:230:14: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  230 |         long p = strtol(optarg, NULL, 10);
      |              ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
../../../../core/src/ndmp/ndmjob_args.c:260:17: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  260 |           char* p;
      |                 ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
../../../../core/src/ndmp/ndmjob_args.c:283:17: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  283 |           char* p = strchr(optarg, '=');
      |                 ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
../../../../core/src/ndmp/ndmjob_args.c:396:13: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  396 |       char* p = strchr(av[c], '=');
      |             ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
../../../../core/src/ndmp/ndmjob_args.c:417:31: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  417 |       struct load_file_entry* p;
      |                               ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
../../../../core/src/ndmp/ndmjob_args.c:431:24: error: declaration of ‘p’ shadows a previous local [-Werror=shadow]
  431 |       char *bp = buf, *p, *ep;
      |                        ^
../../../../core/src/ndmp/ndmjob_args.c:145:9: note: shadowed declaration is here
  145 |   char* p;
      |         ^
cc1: all warnings being treated as errors
gmake[2]: *** [core/src/ndmp/CMakeFiles/ndmjob.dir/build.make:79: core/src/ndmp/CMakeFiles/ndmjob.dir/ndmjob_args.c.o] Error 1
gmake[2]
```

### Environment

```markdown
- OS: Fedora-40
```


### Relevant log output

_No response_

### Relevant traces output

_No response_

### Anything else?

When I helps, I can append the complete build log.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - cats: fix version.map.in [PR #2064]
 - webui: fix for PHP < 7.3 [PR #2067]
 - cmake: fix MARIADB_MYSQL_INSTALL_DB_SCRIPT usage [PR #2040]
+- Fix building ndmjob program [PR #2079]
 
 ### Documentation
 - docs: fix grpc-fd plugin call [PR #2068]
@@ -20,4 +21,5 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [PR #2066]: https://github.com/bareos/bareos/pull/2066
 [PR #2067]: https://github.com/bareos/bareos/pull/2067
 [PR #2068]: https://github.com/bareos/bareos/pull/2068
+[PR #2079]: https://github.com/bareos/bareos/pull/2079
 [unreleased]: https://github.com/bareos/bareos/tree/master

--- a/core/src/ndmp/ndmjob_args.c
+++ b/core/src/ndmp/ndmjob_args.c
@@ -227,11 +227,11 @@ int process_args(int argc, char* argv[])
 
       case 'p': /* -p N -- port number for daemon mode (10000) */
       {
-        long p = strtol(optarg, NULL, 10);
-        if (p < 1 || p > 65535 || (!p && EINVAL == errno)) {
+        long p_l = strtol(optarg, NULL, 10);
+        if (p_l < 1 || p_l > 65535 || (!p_l && EINVAL == errno)) {
           error_byebye("bad -p option");
         }
-        p_ndmp_port = (int)p;
+        p_ndmp_port = (int)p_l;
         break;
       }
 
@@ -257,15 +257,15 @@ int process_args(int argc, char* argv[])
           error_byebye("too many of -E");
         }
         {
-          char* p;
+          char* p_E;
           ndmp9_pval pv;
 
-          p = optarg;
-          pv.name = p;
-          while (*p && *p != '=') p++;
-          if (*p != '=') { error_byebye("missing value in -E"); }
-          *p++ = 0;
-          pv.value = p;
+          p_E = optarg;
+          pv.name = p_E;
+          while (*p_E && *p_E != '=') p++;
+          if (*p_E != '=') { error_byebye("missing value in -E"); }
+          *p_E++ = 0;
+          pv.value = p_E;
           ndma_store_env_list(&E_environment, &pv);
         }
         break;
@@ -280,9 +280,9 @@ int process_args(int argc, char* argv[])
       case 'F': /* -F FILE -- add to list of files */
         if (n_file_arg >= MAX_FILE_ARG) { error_byebye("too many FILE args"); }
         if (strchr(optarg, '=')) {
-          char* p = strchr(optarg, '=');
-          *p++ = 0;
-          file_arg[n_file_arg] = p;
+          char* p_F = strchr(optarg, '=');
+          *p_F++ = 0;
+          file_arg[n_file_arg] = p_F;
           file_arg_new[n_file_arg] = optarg;
           n_file_arg++;
         } else {
@@ -393,9 +393,9 @@ int process_args(int argc, char* argv[])
   for (c = optind; c < ac; c++) {
     if (n_file_arg >= MAX_FILE_ARG) { error_byebye("too many file args"); }
     if (strchr(av[c], '=')) {
-      char* p = strchr(av[c], '=');
-      *p++ = 0;
-      file_arg[n_file_arg] = p;
+      char* p_local = strchr(av[c], '=');
+      *p_local++ = 0;
+      file_arg[n_file_arg] = p_local;
       file_arg_new[n_file_arg] = av[c];
     } else {
       file_arg[n_file_arg] = av[c];
@@ -414,11 +414,11 @@ int process_args(int argc, char* argv[])
 
     /* clean up old load_files_list */
     while (load_files_list) {
-      struct load_file_entry* p;
-      p = load_files_list;
-      load_files_list = p->next;
-      p->next = 0;
-      free(p);
+      struct load_file_entry* p_lfe;
+      p_lfe = load_files_list;
+      load_files_list = p_lfe->next;
+      p_lfe->next = 0;
+      free(p_lfe);
     }
 
     fp = fopen(o_load_files_file, "r");
@@ -428,7 +428,7 @@ int process_args(int argc, char* argv[])
       /* no return */
     }
     while (fgets(buf, sizeof buf, fp) != NULL) {
-      char *bp = buf, *p, *ep;
+      char *bp = buf, *p_loop, *ep;
       int len, slen;
       struct load_file_entry* lfe;
 
@@ -452,18 +452,18 @@ int process_args(int argc, char* argv[])
       lfe->next = 0;
 
       /* see if we have destination */
-      if ((p = strchr(bp, '=')) != 0) {
+      if ((p_loop = strchr(bp, '=')) != 0) {
         int plen;
-        char ch = *p;
+        char ch = *p_loop;
         *p = 0;
 
         /* double conversion -- assume the strings shrink */
-        plen = (p - bp);
-        ndmcstr_to_str(p, &lfe->name[plen + 2], slen - plen - 2);
+        plen = (p_loop - bp);
+        ndmcstr_to_str(p_loop, &lfe->name[plen + 2], slen - plen - 2);
         ndmcstr_to_str(bp, lfe->name, plen + 1);
         file_arg[n_file_arg] = &lfe->name[plen + 2];
         file_arg_new[n_file_arg] = lfe->name;
-        *p = ch;
+        *p_loop = ch;
       } else {
         /* simple conversion copy */
         ndmcstr_to_str(bp, lfe->name, slen - 1);

